src/prefect/tasks/shell.py

```diff
@@ -107,9 +112,11 @@ def run(self, command: str = None, env: dict = None) -> str:
             sub_process.wait()
             if sub_process.returncode:
                 msg = "Command failed with exit code {0}: {1}".format(
-                    sub_process.returncode, line
+                    sub_process.returncode,
+                    "\n".join(lines) if self.log_stderr else line,
```

Comment:
This is about the previous behavior, but what's the reason for logging the last line from `stdout`? This feels like the wrong selection to me (in the case of multi-line output, this will look strangely cropped), and depending on the CLI may leak something secret if the user wasn't expecting it to be logged to cloud.

Also note that if no stdout is sent `line` will be `None`.

Reply:
Ah okay good call, will make that change